### PR TITLE
feat: add an instance metadata experimental service

### PR DESCRIPTION
**Pull request by aramissennyeydd**
## Hey, I just made a Pull Request!

Work towards #25431. This adds a new `instanceMetadataService` that exposes the set of locally installed features. This is the first part of a better discovery system for split backends.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-plugin-api/src/alpha.ts [hunk 1]
```diff
@@ -34,3 +34,19 @@ export const featureDiscoveryServiceRef =
     id: 'core.featureDiscovery',
     scope: 'root',
   });
+
+/**
+ * EXPERIMENTAL: Instance metadata service.
+ *
+ * @alpha
+ */
+export const EXPERIMENTAL_instanceMetadataServiceRef = createServiceRef<
```
Comment (Rugvip) [inline packages/backend-plugin-api/src/alpha.ts:43]: No need to also use the `EXERIMENTAL_` prefix here, it's already in alpha

File: packages/backend/src/instanceMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * limitations under the License.
+ */
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { EXPERIMENTAL_instanceMetadataServiceRef } from '@backstage/backend-plugin-api/alpha';
+
+// Example usage of the instance metadata service to log the installed features.
+export default createBackendPlugin({
```
Comment (Rugvip) [inline packages/backend/src/instanceMetadata.ts:23]: Happy to have this one installed by default too 😁

**Review by Rugvip — COMMENTED** [review]
Nice! 👍 

Just minor stuff and we can :shipit:

File: packages/backend-plugin-api/src/alpha.ts (hunk 1 above)
Comment (aramissennyeydd) [reply packages/backend-plugin-api/src/alpha.ts]: updated in [c1a3bdb](https://github.com/backstage/backstage/pull/27921/commits/c1a3bdbe1276f7a0a88861f41eff00662b2d9bc5)

File: packages/backend/src/instanceMetadata.ts (hunk 2 above)
Comment (aramissennyeydd) [reply packages/backend/src/instanceMetadata.ts:23]: done in [c1a3bdb](https://github.com/backstage/backstage/pull/27921/commits/c1a3bdbe1276f7a0a88861f41eff00662b2d9bc5) 😁

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Let's :shipit:

**Pull request merged**